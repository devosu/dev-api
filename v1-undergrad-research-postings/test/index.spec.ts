// test/index.spec.ts

// Testing essential imports.
import { SELF } from "cloudflare:test";
import { describe, expect, it } from "vitest";

// Third-party validator import.
import { type ResearchPosting, ResearchPostingSchema } from "@keminghe/osu";

// -----------------------------------------------------------------------------
// Integration test suite for the v1-undergrad-research-postings worker.
describe("The v1-undergrad-research-postings worker", () => {
  it("(integration) responds to GET requests with a valid list of all undergrad research postings at OSU.", async () => {
    const response = await SELF.fetch("https://example.com", { method: "GET" });

    // Verify success status.
    expect(response.status).toBe(200);

    // Validate the response body/data against the schema.
    const postings: ResearchPosting[] = await response.json();
    ResearchPostingSchema.array().parse(postings);
  });

  it("(integration) responds to non-GET requests with 418 I'm a teapot.", async () => {
    const response = await SELF.fetch("https://example.com", {
      method: "POST",
    });
    expect(response.status).toBe(418);
    expect(response).toMatchSnapshot();
  });
});
