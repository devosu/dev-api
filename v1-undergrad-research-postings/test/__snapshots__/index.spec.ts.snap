// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The v1-undergrad-research-postings worker > (integration) responds to non-GET requests with 418 I'm a teapot. 1`] = `Response {}`;
